scripts/run-local-perf-test.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+SCRIPT_BASE="$(cd -P "$(dirname "$0")" && pwd)"
+
+# Usage:
+# run-local-smoke-test.sh
+
+# Either have these environment variables set, or change the script below (don't commit that!)
+
+# export PORT_CLIENT_ID=""
+# export PORT_CLIENT_SECRET=""
+# export PORT_BASE_URL=http://localhost:3000
+export VERBOSE=1
+
+export SMOKE_TEST_SUFFIX="${SMOKE_TEST_SUFFIX:-perf-${RANDOM}}"
+export OCEAN__INTEGRATION__CONFIG__ENTITY_AMOUNT_SEED_START=${ENTITY_AMOUNT_SEED_START:--1}
+export OCEAN__INTEGRATION__CONFIG__ENTITY_AMOUNT_SEED_END=${ENTITY_AMOUNT_SEED_END:--1}
+export OCEAN__INTEGRATION__CONFIG__ENTITY_KB_SIZE_FACTOR=${ENTITY_KB_SIZE_FACTOR:--1}
+export OCEAN__INTEGRATION__CONFIG__THIRD_PARTY_BATCH_SIZE=${THIRD_PARTY_BATCH_SIZE:--1}
+export OCEAN__INTEGRATION__CONFIG__THIRD_PARTY_LATENCY_FACTOR=${THIRD_PARTY_LATENCY_FACTOR:--1}
+
+LOG_FILE_MD="${SCRIPT_BASE}/../perf-test-results-${SMOKE_TEST_SUFFIX}.log.md"
+
+echo "Running perf test with ${ENTITY_AMOUNT_SEED_START}-${ENTITY_AMOUNT_SEED_END} entities per department"
+echo "Entity KB size factor: ${ENTITY_KB_SIZE_FACTOR}"
+echo "Third party: Batch ${THIRD_PARTY_BATCH_SIZE} Latency ${THIRD_PARTY_LATENCY_FACTOR}"
+
+
+_log(){
+    echo "| $(date -u +%H:%M:%S) | ${1} |" >> "${LOG_FILE_MD}"
+    echo "${1}"
+}
+
+_run() {
+    RUN_TYPE=${1}
+    START_NS=$(date +%s%N)
+    _log "Starting ${RUN_TYPE}"
+    RUN_LOG_FILE="./${RUN_TYPE}.log"
+    time "${SCRIPT_BASE}/run-local-smoke-test.sh" | tee "${RUN_LOG_FILE}"
+    END_NS=$(date +%s%N)
+    ELAPSED_MS=$(((END_NS - START_NS) / 1000000))
+    _log "${RUN_TYPE}: Duration $((ELAPSED_MS / 1000)) seconds"
+    echo "Trying to grep"
+    set -x
+    ls -lsa "${RUN_LOG_FILE}"
+    tail -n 15 "${RUN_LOG_FILE}"
+    head -n 5 "${RUN_LOG_FILE}"
+    echo "GREEEEPPPPP"
+    grep -P '\| Upserting' "${RUN_LOG_FILE}" || echo "Failed"
+    set +x
+    _log "${RUN_TYPE}: $(grep '| Upserting' "${RUN_LOG_FILE}" | tail -n 1 | awk -F\| '{print $3};' | xargs)"
+    DELETED=$(grep '| Deleting' "${RUN_LOG_FILE}" | tail -n 1 | awk -F\| '{print $3};' | xargs)
+    if [[ -n "${DELETED}" ]]; then
+        _log "${RUN_TYPE}: ${DELETED}"
+    fi
+
+}
```

Comment:
what is going here? why we are not just writing results to a file and then printing?

Reply:
isn't there a library that can be used to help us

Reply:
Managed to make it work with ruby